examples/gadgets/datasource/fields/main.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	ocihandler "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+)
+
+func do() error {
+	const opPriority = 50000
+	myOperator := simple.New("myHandler", simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+		for _, d := range gadgetCtx.GetDataSources() {
+			// Access some specific fields
+			pidF := d.GetField("pid")
+			commF := d.GetField("comm")
+			fnameF := d.GetField("fname")
+
+			d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				pid := pidF.Uint32(data)
```

Comment:
How does developer know it's an `uint32`? I think we are missing a helper here to be able to make it more generic:

```go
	pid := new(pidF.Type())
	pidF.Unmarshal(data, pid)
```

Does it make sense to you? Do you think that would be possible?

cc @flyth

Reply:
@flyth suggested to have a `Any()` or similar accessor method in https://github.com/inspektor-gadget/inspektor-gadget/pull/2753#discussion_r1575290682. In this case I think we're missing documentation, there should be a place that mentions all the fields that are provided by a gadget, their meaning and type. Perhaps they can be extracted from the metadata file and exposed in a markdown?, or let the user check the metadata file directly.